dev-tools/reconfigurator-cli/src/lib.rs

```diff
@@ -320,6 +330,53 @@ struct SledArgs {
     filter: SledFilter,
 }
 
+#[derive(Debug, Args)]
+struct SledSetPolicyArgs {
+    /// id of the sled
+    sled_id: SledUuid,
+
+    /// The policy to set for the sled
+    #[clap(value_enum)]
+    policy: SledPolicyOpt,
+}
+
+#[derive(Clone, Copy, Debug, ValueEnum)]
+enum SledPolicyOpt {
+    InService,
+    NonProvisionable,
+    Expunged,
+}
+
+impl fmt::Display for SledPolicyOpt {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            SledPolicyOpt::InService => write!(f, "in-service"),
+            SledPolicyOpt::NonProvisionable => write!(f, "non-provisionable"),
+            SledPolicyOpt::Expunged => write!(f, "expunged"),
+        }
+    }
+}
+
+impl From<SledPolicyOpt> for SledPolicy {
+    fn from(value: SledPolicyOpt) -> Self {
+        match value {
+            SledPolicyOpt::InService => SledPolicy::InService {
+                provision_policy: SledProvisionPolicy::Provisionable,
+            },
+            SledPolicyOpt::NonProvisionable => SledPolicy::InService {
+                provision_policy: SledProvisionPolicy::NonProvisionable,
+            },
+            SledPolicyOpt::Expunged => SledPolicy::Expunged,
+        }
+    }
+}
```

Comment:
I wrote this but honestly didn't end up needing it -- left it in here because it seems somewhat useful.